src/client/clientmap.cpp

```diff
@@ -386,62 +405,89 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
 						material.setFlag(video::EMF_WIREFRAME,
 							m_control.show_wireframe);
 
-						drawbufs.add(buf, block_pos, layer);
+						if (is_transparent_pass) {
+							// Same comparison as in MeshBufListList
+							bool new_material = material.getTexture(0) != previous_material.getTexture(0) ||
+									material != previous_material;
+
+							draw_order.emplace_back(block_pos, buf, !new_material);
+
+							if (new_material)
+								previous_material = material;
+						}
+						else
+							grouped_buffers.add(buf, block_pos, layer);
```

Comment:
Indentation looks wrong here

Reply:
Not sure what you mean. Github renders 2 spaces/tab, and new_material expression spans two lines => double indent. All other statements have a single indent, braces are only used when necessary.

Reply:
Right, will be the latter then. The else-clause should have braces too.

Reply:
Done now